src/analysis/processing/qgsalgorithmrandompointsextent.cpp

```diff
@@ -0,0 +1,180 @@
+/***************************************************************************
+                         qgsalgorithmrandompointsextent.cpp
+                         ---------------------
+    begin                : November 2019
+    copyright            : (C) 2019 by Clemens Raffler
+    email                : clemens dot raffler at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+//Disclaimer: The algorithm optimizes the original Random points in extent algorithm, (C) Alexander Bruy, 2014
+
+#include "qgsalgorithmrandompointsextent.h"
+#include "random"
+
+///@cond PRIVATE
+
+QString QgsRandomPointsExtentAlgorithm::name() const
+{
+  return QStringLiteral( "randompointsinextent" );
+}
+
+QString QgsRandomPointsExtentAlgorithm::displayName() const
+{
+  return QObject::tr( "Random points in extent" );
+}
+
+QStringList QgsRandomPointsExtentAlgorithm::tags() const
+{
+  return QObject::tr( "random,points,extent,create" ).split( ',' );
+}
+
+QString QgsRandomPointsExtentAlgorithm::group() const
+{
+  return QObject::tr( "Vector creation" );
+}
+
+QString QgsRandomPointsExtentAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectorcreation" );
+}
+
+void QgsRandomPointsExtentAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterExtent( QStringLiteral( "EXTENT" ), QObject::tr( "Input extent" ) ) );
+  addParameter( new QgsProcessingParameterNumber( QStringLiteral( "NUM_POINTS" ), QStringLiteral( "Number of points" ), QgsProcessingParameterNumber::Integer, 1, false, 1 ) );
+  addParameter( new QgsProcessingParameterCrs( QStringLiteral( "CRS" ), QObject::tr( "Target CRS" ), QStringLiteral( "ProjectCrs" ) ) );
```

Comment:
Changing parameter name is an API break, as it will break existing scripts and models.

Reply:
We could possibly make this an optional hidden parameter now, and try to take the crs from the "extent" parameter instead.

Reply:
@nyalldawson I tried hiding the parameter from the user with the `FlagHidden` enum but this seems to make the crs parameter also invisible for the `QgsProcessingParameterDistance` (resulting in unknown units for this parameter). This is due to `QgsProcessingParameterDistance` not being able to get CRSs from extent parameters.  I declared it now as an advanced parameter as workaround.

Reply:
Ok, let's leave it for now. I'd move it back from being an advanced parameter if we're still actively using it.

Reply:
There's just this point remaining ^^ and we can merge

Reply:
Now we can merge ;)